src/server.rs

```diff
@@ -350,11 +289,11 @@ pub async fn run(
     block_watch_dir: impl AsRef<Path>,
     mut state: IndexerState,
     phase_sender: Arc<watch::Sender<MinaIndexerRunPhase>>,
-    mut query_receiver: mpsc::Receiver<(
+    mut query_receiver: mpsc::UnboundedReceiver<(
```

Comment:
Using an unbounded array for IPC is dubious. I would prefer to see it bounded but something more reasonable than `50` like `1024`. Non-blocking